### PR TITLE
CORE-171 protocol handshake response

### PR DESCRIPTION
**Pull request by rabbitonweb**
1. Adds encryption to second phase protocol handshake (both request and response)
2. Covers all cases in tests (some left pending for discussion)
3. Fixes & refactoring to existing code

### CODE REVIEW

File: node/src/main/scala/coop/rchain/node/main.scala [hunk 1]
```diff
@@ -111,6 +111,15 @@ object Main {
       def generateNonce: Task[Nonce] = Task.delay {
         encoder.decode("69696ee955b62b73cd62bda875fc73d68219e0036b7a0b37")
       }
+
+      def encrypt(pub: Key, sec: Key, nonce: Nonce, message: Array[Byte]): Task[Array[Byte]] =
+        Task.delay {
+          message
```
Comment (henrytill) [inline node/src/main/scala/coop/rchain/node/main.scala]: What's going on here?

File: node/src/main/scala/coop/rchain/node/main.scala [hunk 2]
```diff
@@ -208,6 +188,37 @@ object Main {
       def toEffect: Effect[A] = t.liftM[CommErrT]
     }
 
+    val net = new UnicastNetwork(src, Some(p2p.Network))
```
Comment (henrytill) [inline node/src/main/scala/coop/rchain/node/main.scala]: No need to `new`, yah?

File: comm/src/main/scala/coop/rchain/p2p/p2p.scala [hunk 3]
```diff
@@ -157,4 +185,48 @@ object Network extends ProtocolDispatcher[java.net.SocketAddress] {
...
+      frameable: Nonce => Frameable)(implicit keysStore: Kvs[F, PeerNode, Array[Byte]],
+                                     err: ApplicativeError_[F, CommError]): F[FrameMessage] =
+    for {
+      local <- Communication[F].local
+      keys  <- Encryption[F].fetchKeys
+      nonce <- Encryption[F].generateNonce
+      remotePubKey <- keysStore
+                       .get(remote)
+                       .map(k => err.fromEither(k.toRight(publicKeyNotAvailable(remote))))
+                       .flatten
```
[the remark this answers is not in the document]
Comment (rabbitonweb) [reply comm/src/main/scala/coop/rchain/p2p/p2p.scala]: 👍

File: comm/src/main/scala/coop/rchain/p2p/p2p.scala [hunk 4]
```diff
@@ -157,4 +185,48 @@ object Network extends ProtocolDispatcher[java.net.SocketAddress] {
...
```
[the remark this answers is not in the document]
Comment (rabbitonweb) [reply comm/src/main/scala/coop/rchain/p2p/p2p.scala]: 👍

File: node/src/main/scala/coop/rchain/node/main.scala (hunk 1 above)
Comment (rabbitonweb) [reply node/src/main/scala/coop/rchain/node/main.scala]: `//TODO FIX-ME temp implemmentation, production shoul use crypto module` should tell ou everything :)

This is mock implementation. Adding real encryption is part of next PR https://github.com/rchain/rchain/pull/410/files#diff-99c215e4c2e53bf45b4887028939a176L100

File: node/src/main/scala/coop/rchain/node/main.scala (hunk 2 above)
Comment (rabbitonweb) [reply node/src/main/scala/coop/rchain/node/main.scala]: well, this should not really be case class, but i will remove `new` for now

Comment (henrytill) [reply node/src/main/scala/coop/rchain/node/main.scala]: yeah, one or the other, i guess

File: node/src/main/scala/coop/rchain/node/main.scala (hunk 1 above)
Comment (henrytill) [reply node/src/main/scala/coop/rchain/node/main.scala]: gotcha

File: node/src/main/scala/coop/rchain/node/main.scala (hunk 2 above)
Comment (rabbitonweb) [reply node/src/main/scala/coop/rchain/node/main.scala]: I've ended up changing `UnicastNetwork` to be normal class, it never should be case class in the first place.

**Comment by rabbitonweb** [comment]
this is ready for review one more time :)

**Pull request merged**